### PR TITLE
Fixes #151 DomainOfExpertise from session

### LINKED ISSUE #151: Session.QueryCurrentDomainOfExpertise and QueryDomainOfExpertise should have an Iteration as a parameter
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the CDP4-SDK
- [x] I have searched [open](https://github.com/RHEAGROUP/CDP4-SDK-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/CDP4-SDK-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
A user can open the same Iteration multiple times using other different DomainOfExpertises. 
In that case the the OpenIterations property returns more than 1 non-frozen iterations. 

The Session.QueryCurrentDomainOfExpertise method however expects a single, or no non-frozen iterations to be found in the OpenIterations property.
The Session.QueryCurrentDomainOfExpertise method should expect a Iteration parameter and get the correct data using that Iteration instead of search for a non-frozen iteration.
The same for Session.QueryDomainOfExpertise().

### PR DESCRIPTION
**Pull request by lxatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-SDK-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-SDK [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-SDK-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Bugfix and refactoring DomainOfExpertise from Session/ISession

### CODE REVIEW

File: CDP4Dal/CDP4Dal.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	<TargetFrameworks>net45;net451;net452;net46;net461;net462;net47;net471;net472;net48;netstandard2.0;netstandard2.1;netcoreapp3.1</TargetFrameworks>
     <Company>RHEA System S.A.</Company>
     <Title>CDP4Dal Community Edition</Title>
-    <VersionPrefix>6.3.0</VersionPrefix>
+    <VersionPrefix>6.3.1</VersionPrefix>
```
Comment (alexatrhea) [inline CDP4Dal/CDP4Dal.csproj:7]: This is a non-backwards compatible change. I think this will be SDK 7.0.0.

**Review by alexatrhea — CHANGES_REQUESTED** [review]

File: CDP4Dal/ISession.cs [hunk 2]
```diff
@@ -312,20 +312,13 @@ public interface ISession
...
-        /// <summary>
-        /// Queries the current <see cref="DomainOfExpertise"/> from the session for the current <see cref="Iteration"/>
-        /// </summary>
-        /// <returns>
-        /// The <see cref="DomainOfExpertise"/> if selected, null otherwise.
-        /// </returns>
-        DomainOfExpertise QueryCurrentDomainOfExpertise();
-
         /// <summary>
         /// Queries the <see cref="Participant"/>'s <see cref="DomainOfExpertise"/>'s from the session for the current <see cref="Iteration"/>
```
Comment (samatrhea) [inline CDP4Dal/ISession.cs:316]: please update the docs -> current iteration needs to become `provided iteration` and the `returns` statement needs to be updated as well

File: CDP4Dal/Session.cs [hunk 3]
```diff
@@ -948,33 +948,16 @@ private void AddIterationToOpenList(Guid iterationId, DomainOfExpertise activeDo
             this.AddRdlToOpenList(modelRdl);
         }
 
-        /// <summary>
-        /// Queries the current <see cref="DomainOfExpertise"/> from the session for the current <see cref="Iteration"/>
-        /// </summary>
-        /// <returns>
-        /// The <see cref="DomainOfExpertise"/> if selected, null otherwise.
-        /// </returns>
-        public DomainOfExpertise QueryCurrentDomainOfExpertise()
```
Comment (samatrhea) [inline CDP4Dal/Session.cs:957]: i think there was indeed already a method called `QuerySelectedDomainOfExpertise(Iteration iteration)` that was taking care of this, and indeed, it needs an `Iteration` as parameter

File: CDP4Dal/ISession.cs [hunk 4]
```diff
@@ -312,20 +312,13 @@ public interface ISession
...
         /// </summary>
+        /// <param name="iteration">The <see cref="Iteration"/></param>
         /// <returns>
         /// The <see cref="DomainOfExpertise"/> if selected, null otherwise.
         /// </returns>
-        IEnumerable<DomainOfExpertise> QueryDomainOfExpertise();
+        IEnumerable<DomainOfExpertise> QueryDomainOfExpertise(Iteration iteration);
```
Comment (samatrhea) [inline CDP4Dal/ISession.cs:322]: in the source file, can this method be placed after the existing `QuerySelectedDomainOfExpertise`, then they are close to eachother, should be easier to maintain the code in the future.

**Review by samatrhea — CHANGES_REQUESTED** [review]

File: CDP4Dal/ISession.cs [hunk 5]
```diff
@@ -128,6 +128,15 @@ public interface ISession
         /// A <see cref="DomainOfExpertise"/> if has been selected for the <see cref="Iteration"/>, null otherwise.
         /// </returns>
         DomainOfExpertise QuerySelectedDomainOfExpertise(Iteration iteration);
+        
+        /// <summary>
+        /// Queries the <see cref="Participant"/>'s <see cref="DomainOfExpertise"/>'s from the session for the provided <see cref="Iteration"/>
+        /// </summary>
+        /// <param name="iteration">The <see cref="Iteration"/></param>
+        /// <returns>
+        /// The <see cref="DomainOfExpertise"/> if selected, null otherwise.
```
Comment (samatrhea) [inline CDP4Dal/ISession.cs:137]: this is incorrect

File: CDP4Dal/Session.cs [hunk 6]
```diff
@@ -225,6 +225,26 @@ public DomainOfExpertise QuerySelectedDomainOfExpertise(Iteration iteration)
 
             return iterationDomainPair.Value == null || iterationDomainPair.Value.Item1 == null ? null : iterationDomainPair.Value.Item1;
         }
+        
+        /// <summary>
+        /// Queries the <see cref="Participant"/>'s <see cref="DomainOfExpertise"/>'s from the session for the provided <see cref="Iteration"/>
+        /// </summary>
+        /// <returns>
+        /// <param name="iteration">The <see cref="Iteration"/></param>
+        /// The <see cref="DomainOfExpertise"/> if selected, null otherwise.
```
Comment (samatrhea) [inline CDP4Dal/Session.cs:234]: this is incorrect

**Review by samatrhea — CHANGES_REQUESTED** [review]
the docs on returns seem to be incorrect

**Review by samatrhea — APPROVED** [review]

**Review by alexatrhea — APPROVED** [review]

**Pull request merged**